test/Microsoft.AspNetCore.Mvc.FunctionalTests/MvcTestFixture.cs

```diff
@@ -58,12 +55,26 @@ public void Dispose()
             _server.Dispose();
         }
 
-        private static string GetApplicationPath(string relativePath)
+        private static string GetApplicationPath(string solutionRelativePath)
         {
             var startupAssembly = typeof(TStartup).GetTypeInfo().Assembly;
             var applicationName = startupAssembly.GetName().Name;
             var applicationBasePath = PlatformServices.Default.Application.ApplicationBasePath;
-            return Path.GetFullPath(Path.Combine(applicationBasePath, relativePath, applicationName));
+
+            var directoryInfo = new DirectoryInfo(applicationBasePath);
+            do
+            {
+                var slnFile = new FileInfo(Path.Combine(directoryInfo.FullName, SolutionName));
```

Comment:
Since it's `SolutionName`, use `solutionFile` or perhaps `fileInfo` / `solution[File]Info` to align w/ `directoryInfo`. I know: Too many choices but `slnFile` doesn't match anything else here and is an abbrev.